test/MusicbrainzCDID.cxx

```diff
@@ -0,0 +1,234 @@
+#include "fs/AllocatedPath.hxx"
+#include "lib/cdio/Paranoia.hxx"
+#include "util/StringCompare.hxx"
+#include "util/StringSplit.hxx"
+#include "util/NumberParser.hxx"
+#include "util/ScopeExit.hxx"
+#include "util/PrintException.hxx"
+#include <cdio/cd_types.h>
+#include <cstdint>
+#include <vector>
+
+extern "C" {
+#include <libavutil/sha.h>
+#include <libavutil/base64.h>
+#include <libavutil/mem.h>
+}
+
+#include <stdio.h>
+
+using std::string_view_literals::operator""sv;
+
+struct TrackOffsets
+{
+	int firstTrackNumber = 1;
+	int lastTrackNumber = -1;
+	int leadIn = -1;
+	std::vector<int> frameOffsets;
+
+	bool isValid () const
+	{
+		return lastTrackNumber >= firstTrackNumber
+			&& leadIn > 0
+			&& frameOffsets.size() > 0;
+	}
+};
+
+static const std::string
+makeMusicBrainzIdWithOffsets (const TrackOffsets& trackOffsets)
+{
+	const size_t numBitsSha1 = 160;
+	const size_t digestSize = numBitsSha1 / 8;
+	uint8_t digest[digestSize];
```

Comment:
This STILL is a variable-length array. Please do not do that, it's illegal in C++!